avm/ptn/sa/conversation-knowledge-mining/main.bicep

```diff
@@ -0,0 +1,1168 @@
+param functionsManagedEnvironmentLocation string = solutionLocation
+@description('Optional. Location for the Key Vault resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param keyVaultLocation string = solutionLocation
+@description('Optional. Location for the Log Analytics Workspace resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param logAnalyticsWorkspaceLocation string = solutionLocation
+@description('Optional. Location for the Managed Identity resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param managedIdentityLocation string = solutionLocation
+@description('Optional. Location for the Script Copy Data resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param scriptCopyDataLocation string = solutionLocation
+@description('Optional. Location for the Script Index Data resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param scriptIndexDataLocation string = solutionLocation
+@description('Optional. Location for the SQL Server resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param sqlServerLocation string = databasesLocation
+@description('Optional. Location for the Storage Account resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param storageAccountLocation string = solutionLocation
+@description('Optional. Location for the Web App resource deployment.')
+@metadata({ azd: { type: 'location' } })
+param webAppLocation string = solutionLocation
+
+// PARAMETERS: Log Analytics workspace configuration
+@description('Optional. The SKU for the Log Analytics Workspace. If empty, PerGB2018 will be used.')
+@allowed(['CapacityReservation', 'Free', 'LACluster', 'PerGB2018', 'PerNode', 'Premium', 'Standalone', 'Standard'])
+param logAnalyticsWorkspaceSkuName string = 'PerGB2018'
+
+@description('Optional. The number of days to retain the data in the Log Analytics Workspace. If empty, it will be set to 30 days.')
+@minValue(0)
+@maxValue(730)
+param logAnalyticsWorkspaceDataRetentionInDays int = 30
+
+// PARAMETERS: Key Vault configuration
+@allowed(['premium', 'standard'])
+@description('Optional. The SKU for the Key Vault. If empty, standard will be used.')
+param keyVaultSku string = 'standard'
+
+@description('Optional. The Key Vault create mode. Indicates whether the vault need to be recovered from purge or not. If empty, default will be used.')
+@allowed(['default', 'recover'])
+param keyVaultCreateMode string = 'default'
+
+@description('Optional. If set to true, The Key Vault soft delete will be enabled.')
+param keyVaultSoftDeleteEnabled bool = true
+
+@description('Optional. The number of days to retain the soft deleted vault. If empty, it will be set to 7.')
+@minValue(7)
+@maxValue(90)
+param keyVaultSoftDeleteRetentionInDays int = 7
+
+@description('Optional. If set to true, The Key Vault purge protection will be enabled. If empty, it will be set to false.')
+param keyVaultPurgeProtectionEnabled bool = false
```

Comment:
Maybe not for version 1, but down the line I'd highly recommend to cluster properties together using UDTs. For example one parameter `param keyVaultConfiguration keyVaultConfigurationType` to cluster these together. Same for all the name & location parameters which could be combined into one parameter each.

That being said - UDTs **do** contribute to a template's size. So there's a catch.